src/NaturalLanguage/NLStrongDictionary.cs

```diff
@@ -19,11 +21,16 @@ public NLStrongDictionary (NSDictionary dictionary) : base (dictionary)
 
 		public string[] this [NSString key] {
 			get {
-				if (key == null)
-					throw new ArgumentNullException (nameof (key));
+				if (key is null)
+					ObjCRuntime.ThrowHelper.ThrowArgumentNullException (nameof (key));
 
 				var value = CFDictionary.GetValue (Dictionary.Handle, key.Handle);
-				return CFArray.StringArrayFromHandle (value);
+				var array = CFArray.StringArrayFromHandle (value) ?? throw new ArgumentOutOfRangeException (nameof (key));
+				foreach (var str in array) {
+					if (str is null)
+						ObjCRuntime.ThrowHelper.ThrowArgumentNullException ($"Key value {nameof (key)} contains a null string.");
+				}
+				return array!;
```

Comment:
`CFDictionary.GetValue (Dictionary.Handle, key.Handle)` returns a string?[]? but the setter can only handle a string[] so I thought it would be good to have these checks here

Reply:
It's quite common that a dictionary returns null for a key that doesn't exist, but that doesn't mean null is a valid value for a given key.

I would:

* Change the indexer's type to `string?[]?`
* Keep returning null in the getter
* Change the setter to throw ArgumentNullException if a null value is given (maybe SetArrayValue already does that?)

Reply:
+1 I had the same comment but I did not push the button 🤦

Reply:
@rolfbjarne @mandel-macaque 
I was trailing the method calls from the setter and everything works out well if we pass in a null such as
`dictionary[key] = null;`

But when we pass in something like this:
`dictionary[key] = new string[] {null};`
It goes down a few levels to here:
https://github.com/xamarin/xamarin-macios/blob/main/src/Foundation/NSArray.cs#L213
where it sets `val = NSNull.Null.Handle;` which I assume is okay?

Should I leave the `SetArrayValue (key, value);` as `SetArrayValue (key, value!);` or go through a loop and throw an exception if we have any nulls in the string array we are setting the key value to?

Reply:
I would say just leave it as `SetArrayValue (key, value!);`, since that's the safest (it's the current behavior), and it's also the simplest (which is always good).